maggy/ablation/ablationstudy.py

```diff
@@ -0,0 +1,231 @@
+        self.label_name = label_name
+        self.custom_dataset_generator = kwargs.get('dataset_generator', False)
+
+    def to_dict(self):
+        """
+        Returns the ablation study configuration as a Python dictionary.
+        :return: A dictionary with ablation study configuration parameters as keys
+        (i.e. 'training_dataset_name', 'included_features', etc.)
+        :rtype: dict
+        """
+        ablation_dict = {
+            'training_dataset_name': self.hops_training_dataset_name,
+            'training_dataset_version': self.hops_training_dataset_version,
+            'label_name': self.label_name,
+            'included_features': list(self.features.included_features),
+            'included_layers': list(self.model.layers.included_layers),
+            'custom_dataset_generator': True if self.custom_dataset_generator else False,
+        }
+
+        return ablation_dict
+
+
+class Features(object):
+    def __init__(self):
+        self.included_features = set()
+
+    def include(self, *args):
+        """
+        Add one or several features of the dataset to the ablation study. Features can be included one by one or
+        as a list of strings.
+        :param args: Strings or lists of strings, that should match feature names.
+        :type args: str or list
+        :return:
+        """
+        for arg in args:
+            if type(arg) is list:
+                for feature in arg:
+                    self._include_single_feature(feature)
+            else:
+                self._include_single_feature(arg)
+
+    def _include_single_feature(self, feature):
+        if type(feature) is str:
+            self.included_features.add(feature)
+        else:
+            raise ValueError(
+                "features.include() only accepts strings or lists of strings, "
+                "but it received {0} which is of type '{1}'."
+                .format(str(feature), type(feature).__name__))
+
+    def exclude(self, *args):
+        """
+        Exclude one or several features of the dataset from the list of features that have been included in the
+        ablation study. Features can be excluded one by one or as a list of strings. This method will check to see
+        if the feature has already been included in the study before excluding it.
+        :param args: Strings or lists of strings, that should match feature names.
+        :type args: str or list
+        :return:
+        """
+        for arg in args:
+            if type(arg) is list:
+                for feature in arg:
+                    self._exclude_single_feature(feature)
+            else:
+                self._exclude_single_feature(arg)
+
+    def _exclude_single_feature(self, feature):
+        if type(feature) is str:
+            if feature in self.included_features:
+                self.included_features.remove(feature)
+                print("Feature '{0}' is excluded from the ablation study.")
```

Comment:
is not `.format(str(feature))` missing here?

Reply:
indeed! :D

Reply:
Fixed with
https://github.com/logicalclocks/maggy/pull/27/commits/05e4efa4894526ed2e275b34929d4593bbad6cb9